src/validation.cpp

```diff
@@ -5072,6 +5080,13 @@ bool DumpMempool(const CTxMemPool& pool)
         }
 
         file << mapDeltas;
+
+        {
+            LOCK(pool.cs);
+            LogPrintf("Writing %d unbroadcast transactions to disk.\n", pool.m_unbroadcast_txids.size());
+            file << pool.m_unbroadcast_txids;
```

Comment:
This is racy and you might end up with zombie txids here that are never removed because the corresponding mempool tx has not been written to disk

Reply:
I agree. All of the data should be brought out of the mempool under one lock. Make a local `std::set<uint256>` and populate it under the same lock as `mapDeltas` and `vinfo` get populated.

Reply:
oops. thanks for catching. fixed